streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import org.apache.kafka.streams.kstream.internals.Change;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
+
+public class KTableSuppressProcessor<K, V> implements Processor<K, Change<V>> {
+    private final SuppressedImpl<K, V> suppress;
+    private InternalProcessorContext internalProcessorContext;
+
+    public KTableSuppressProcessor(final SuppressedImpl<K, V> suppress) {
+        this.suppress = suppress;
+    }
+
+    @Override
+    public void init(final ProcessorContext context) {
+        internalProcessorContext = (InternalProcessorContext) context;
+    }
+
+    @Override
+    public void process(final K key, final Change<V> value) {
+        if (suppress.isImmediateEmit() && definedRecordTime(key, value) <= internalProcessorContext.streamTime()) {
```

Comment:
Don't understand the second check? Why must the record-time (or window-end-time) be less-or-equal to current stream-time? For record-time, this should be true all the time, but for window-end time, I don't think this is true -- why do we want to fail for this case?

Reply:
As you pointed out, a record in an open window will have a time greater than the current stream time. This means we should buffer it (when it's implemented) and *not* immediately emit it.

Reply:
Why should we buffer it? If `suppress.isImmediateEmit()` is true, I though we would not buffer it but emit it immediately to obey the config? Or do I miss understand the semantics of `isImmediateEmit()`?

Reply:
hmm I guess my method name is misleading. It's not that the suppression is configured like "emit immediately" (there is no such config option, but maybe there should be).

Rather, it's an internal utility method to indicate whether the buffer config allows us to just emit events that are on-time or late, rather than buffering them.

Regardless, we still need to buffer future events if there is room to buffer them. This logic is just a little murky right now because there isn't a buffer yet.

So this logic is an attempt to emit any event that we can determine right away is legal to emit, since any buffering operation is actually an exception in this PR.

I guess I could have made it simpler by just unilaterally throwing an exception for any record here, but I thought it would be nice to have some non-exceptional paths to have tests for. 

Even though the implementation details will change when we add the buffer, any tests that currently check for events getting immediately emitted should continue to pass on the buffer-based implementation.

Reply:
Ack.